UltimateMedals.as

```diff
@@ -212,7 +216,14 @@ Record@ silver = Record(silverText, 2, -3);
 Record@ bronze = Record(bronzeText, 1, -2);
 Record@ pbest = Record(pbestText, 0, -1, "\\$0ff");
 
-#if TMNEXT||MP4
+#if TMNEXT
+Record@ champion = Record(championText, 5, -6);
+Record@ warrior = Record(warriorText, 6, -7);
+#endif
+
+#if TMNEXT
+array<Record@> times = {champion, warrior, author, gold, silver, bronze, pbest};

```

Comment:
I'm unsure of how these if statements for these extensions work, but I would assume based on the logic here that you would some editing of this array.

Because how will this work if someone doesn't want Champion or Warrior medal support? What if they actually want neither. Will this time show up anyway and does that effect the UI when they dont show up because champion.hidden != true?

Reply:
They can turn of showing them individually, but ultimate medals uses indexes in an array for a few things, and there needs to be a single global array with everything. 
Yes, it's not idea.